### PR TITLE
fix: enhance focus handling for search-box components

### PR DESCRIPTION
**Pull request by louis-bompart**
- Replace the atomic-focus-detector with a `focus` listener and a guard to ensure the newly focused element is _not_ a child of the SearchBoxWrapper main element.
- Handle cross-window focus switching: if we exit the document during a focus out, we discard the event

KIT-3157

### CODE REVIEW

**Review by fbeaudoincoveo — APPROVED** [review]
LGTM, but I unfortunately don't have an iPhone to test it

File: packages/atomic/src/components/common/search-box/search-box-wrapper.tsx [hunk 1]
```diff
@@ -23,7 +24,7 @@ export const SearchBoxWrapper: FunctionalComponent<Props> = (
   };
 
   return (
-    <div part="wrapper" class={getClasses()}>
+    <div part="wrapper" class={getClasses()} onFocusout={props.onFocusOut}>
```
[the remark this answers is not in the document]
Comment (olamothe) [reply packages/atomic/src/components/common/search-box/search-box-wrapper.tsx:27]: `onFocusout` is the actual name of the JSX prop and cannot be changed. 

If anything, it should be modified in `SearchBoxWrapperProps` to be `onFocusout` instead.

**Review by olamothe — APPROVED** [review]
Approved on the condition that we can verify this with iPhone simulator first ;)

**Comment by louis-bompart** [comment]
> Approved on the condition that we can verify this with the iPhone simulator first ;)

The verification has been done ish': I did test with the simulator, but I suspect it didn't test and that there's no way to actually test the behaviour of KIT-1941 without an actual physical device.

**Comment by louis-bompart** [comment]
KIT-3157

**Comment by louis-bompart** [comment]
@lvu285 , can you validate KIT-3299, and a round of testing on focus behavior of the Searchboxes?

You can use the Storybook link that's here https://github.com/coveo/ui-kit/pull/4027#issuecomment-2140746657 and use the Searchbox stories if you prefer :) 

What I tested so far:
 - Type some stuff, tab, ensure it closes
 - Click in searchbox, Alt-Tab, Assert suggestions stays, Alt-Tab back in, Assert suggestions stays and do not flicker.
 - Click in searchbox, click out of window, assert suggestions stays, click back in on the searchbox input, assert suggestions stays
 - Click in searchbox, click out of window, assert suggestions stays, click back not on searchbox input, assert suggestions collapses
 - Focus searchbox, click somewhere else in the page, ensure it closes

**Comment by lvu285** [comment]
It looks ok for me on Storybook, but I'm curious how it would interact with other components too.

**Comment by louis-bompart** [comment]
> It looks ok for me on Storybook, but I'm curious how it would interact with other components too.

Feel free to checkout the branch and run the examples on your machine (yeah, we'll add'em eventually to the demo links :)  baby steps :D )

**Review by olamothe — APPROVED** [review]

**Review by fbeaudoincoveo — APPROVED** [review]

**Pull request merged**